cw_infrastructure/cw_infrastructure/test/validators_test.py

```diff
@@ -0,0 +1,199 @@
+    @mock.patch('cw_infrastructure.check_config_utilities.error',
+                autospec=True)
+    def test_error_max(self, mock_error, max_value, value):
+
+        val = validators.create_integer_range_validator(max_value=max_value)
+        code = val('val', value)
+
+        if value > max_value:
+            # ERROR should be returned, and the error function called
+            self.assertEqual(code, check_config_utilities.ERROR)
+            mock_error.assert_called_once_with('val', mock.ANY)
+        else:
+            # OK should be returned, and the error function not called
+            self.assertEqual(code, check_config_utilities.OK)
+            self.assertFalse(mock_error.called)
+
+    # Test that a warning is raised when values greater than the maximum warning
+    # value are validated
+    @hypothesis.given(max_value=st.integers(), value=st.integers())
+    @mock.patch('cw_infrastructure.check_config_utilities.warning',
+                autospec=True)
+    def test_warn_max(self, mock_warning, max_value, value):
+
+        val = validators.create_integer_range_validator(warn_max_value=max_value)
+        code = val('val', value)
+
+        if value > max_value:
+            # WARNING should be returned, and the warning function called
+            self.assertEqual(code, check_config_utilities.WARNING)
+            mock_warning.assert_called_once_with('val', mock.ANY)
+        else:
+            # OK should be returned, and the warning function not called
+            self.assertEqual(code, check_config_utilities.OK)
+            self.assertFalse(mock_warning.called)
+
+    # Test that ERRORs take priority over WARNINGs when setting minima
+    @mock.patch('cw_infrastructure.check_config_utilities.error',
+                autospec=True)
+    def test_min_error_priority(self, mock_error):
+
+        val = validators.create_integer_range_validator(min_value=1,
+                                                        warn_min_value=2)
+        code = val('val', 0)
+
+        self.assertEqual(code, check_config_utilities.ERROR)
+        mock_error.assert_called_once_with('val', mock.ANY)
+
+    # Test that ERRORs take priority over WARNINGs when setting minima
+    @mock.patch('cw_infrastructure.check_config_utilities.error',
+                autospec=True)
+    def test_max_error_priority(self, mock_error):
+
+        val = validators.create_integer_range_validator(max_value=1,
+                                                        warn_max_value=0)
+        code = val('val', 2)
+
+        self.assertEqual(code, check_config_utilities.ERROR)
+        mock_error.assert_called_once_with('val', mock.ANY)
+
+
+class TestIpAddrValidator(unittest.TestCase):
+
+    # Test that the IP address validator calls into the is_ip_addr
+    # utility function correctly, and correctly handles its result in the
+    # success case.
+    @mock.patch('cw_infrastructure.check_config_utilities.is_ip_addr',
```

Comment:
Why bother mocking this? You can fairly easily provide a bad IP address.

Reply:
Ok, fair enough. Do we have tests for `is_ip_addr` then?

Reply:
Not at the moment - it's a fairly transparent call into a system function. Could be worth doing another layer; mocking the system function and testing it is called correctly?

Reply:
If it's fairly trivial plumbing code then it's probably not worth testing, so this test is sufficient.